File: Cpp/1029_Two_City_Scheduling.cpp
class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) {
        int result = 0;
        int n = costs.size();
        vector<pair<int, int>> difference(n);
        for(int i=0; i<costs.size(); ++i){
            difference[i] = {costs[i][0]-costs[i][1], i};
        }
        sort(difference.begin(), difference.end());
        
        for(int i=0; i<n/2; ++i){
            result += costs[difference[i].second][0];
        }
        for(int i=n/2; i<n; ++i){
            result += costs[difference[i].second][1];
        }
        
        return result;
    }
};

File: Kotlin/1029_Two_City_Scheduling.kt
class Solution {
    fun twoCitySchedCost(costs: Array<IntArray>): Int {
        var result = 0
        
        costs.sortWith(Comparator{a, b -> 
            (a[0] - a[1]).compareTo(b[0] - b[1])
        })
        
        for(i in 0..costs.size / 2 - 1) {
            result += costs[i][0]
            result += costs[i + costs.size / 2][1]
        }

        return result
    }
}